gluten-core/src/main/scala/io/glutenproject/execution/SortExecTransformer.scala

```diff
@@ -269,10 +158,4 @@ object SortExecTransformer {
       case _ => 0
     }
   }
-
-  def needProjection(sortOrders: Seq[SortOrder]): Boolean = {
-    sortOrders
-      .map(_.child)
-      .exists(exp => !exp.isInstanceOf[Attribute] && !exp.isInstanceOf[BoundReference])
```

Comment:
shall we respect `BoundReference` in new method `isNotAttribute` ?

Reply:
I think it's okay to add it, but currently we haven't tested for any corresponding situations.